### PR TITLE
Implement functionality for migration

### PR DESCRIPTION
**Pull request by antban**
Description in #49

### CODE REVIEW

**Comment by v-stepanov** [comment]
But such approach will not allow to migrate partition by partition, right? We will have first to migrate everything and then starting switching leaders.

File: bubuku/features/migrate.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    def __str__(self):
+        return 'Migration links {}, shrink: {}, data_to_move: {}'.format(
+            self.migration,
+            self.shrink,
+            len(self.data_to_migrate) if self.data_to_migrate is not None else 'Unknown')
+
+    def _replace_replicas(self, replicas):
+        replacement = [self.migration[k] for k in replicas if k in self.migration]
+        if self.shrink:
```
Comment (v-stepanov) [inline bubuku/features/migrate.py]: I think I don't get it, what exactly shrink does?

Comment (antban) [inline bubuku/features/migrate.py]: shrink means remove replaced broker ids, So there can be 3 possibilities of usage:

 1. Without shrink: [1, 2, 3] -> [1, 2, 3, 4, 5, 6]
 1. With shrink: [1, 2, 3] -> [4, 5, 6]
 1. Without and than - with: [1, 2, 3] -> [1, 2, 3, 4, 5, 6] -> [4, 5, 6]

**Comment by antban** [comment]
Yes, this approach allows replacement of brokers only. Like if you don't want to have situation like now - terminate and rebalance everything - it's possible to start new instance and than remove old instance usage.

**Comment by v-stepanov** [comment]
:shipit:

**Pull request merged**